tools/src/bin/rust-indexer.rs

```diff
@@ -297,6 +450,14 @@ fn analyze_file(
         }
     };
 
+    let mut source_file = match File::open(file_name.clone()) {
+        Ok(f) => f,
+        Err(_) => {
+            error!("Unable to open source file for snippet purposes: {}", file_name.display());
```

Comment:
When does this happen? And should it really avoid the rest of the analysis from running?

Reply:
As you can tell by the println below, I guess I'm not sure exactly how we want to handle errors.  I was viewing the file not existing as indicative of a major configuration error that should fail fast.  Like how can we be indexing a file if we don't have the source?  (But maybe the fact that all of this analysis is being reverse-engineered from crate analysis stuff means it's possible?)

Reply:
Often times we get analysis data that has "weird" filenames (see e.g. https://github.com/rust-lang/rls/issues/1339) so I think we probably do want to handle this gracefully

Reply:
I'm making this into an Option<> and having us skip the `impls` traversal that depends on this to retrieve the impl name.